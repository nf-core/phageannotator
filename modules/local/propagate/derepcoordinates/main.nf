process PROPAGATE_DEREPCOORDINATES {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta) , path(coordinates)
    tuple val(meta2), path(clusters)

    output:
    tuple val(meta), path("${prefix}_derep_provirus_coords.tsv")    , emit: derep_coords
    path "versions.yml"                                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    derep_coordinates.py \\
        --coordinates ${coordinates} \\
        --clusters ${clusters} \\
        --output ${prefix}_derep_provirus_coords.tsv \\
        ${args}


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        numpy: \$(echo \$(numpy_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_derep_provirus_coords.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        numpy: \$(echo \$(numpy_version.py 2>&1))
    END_VERSIONS
    """
}
