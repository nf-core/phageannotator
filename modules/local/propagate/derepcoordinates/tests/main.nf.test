nextflow_process {

    name "Test Process: PROPAGATE_DEREPCOORDINATES"
    script "../main.nf"
    process "PROPAGATE_DEREPCOORDINATES"

    setup {
        run("PROPAGATE_EXTRACTPROVIRUSES") {
            script "../../extractproviruses/main.nf"
            process {
                """
                input[0] = Channel.of(
                        [
                            [ id: 'test' ],
                            file(params.pipelines_testdata_base_path + "modules/local/propagate/propagate_test_contigs.fasta.gz", checkIfExists: true)
                        ]
                    )
                    input[1] = Channel.of(
                        [
                            [ id: 'test' ],
                            file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilis_genomad_virus_summary.tsv", checkIfExists: true)
                        ]
                    )
                    input[2] = Channel.of(
                        [
                            [ id: 'test' ],
                            file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilis_checkv_contamination.tsv", checkIfExists: true)
                        ]
                    )
                """
            }
        }
    }


    test("coords.tsv + clusters.tsv") {

        when {
            process {
                """
                input[0] = PROPAGATE_EXTRACTPROVIRUSES.out.provirus_coords
                input[1] = Channel.of(
                    [
                        [ id: 'test' ],
                        file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilis_clusters.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("coords.tsv + clusters.tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = PROPAGATE_EXTRACTPROVIRUSES.out.provirus_coords
                input[1] = Channel.of(
                    [
                        [ id: 'test' ],
                        file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilis_clusters.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
