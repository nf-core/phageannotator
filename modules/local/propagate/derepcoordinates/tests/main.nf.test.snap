{
    "coords.tsv + clusters.tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_derep_provirus_coords.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,244bc446383cbd763a9b5a5d58208fb9"
                ],
                "derep_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_derep_provirus_coords.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,244bc446383cbd763a9b5a5d58208fb9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T08:42:07.920114246"
    },
    "coords.tsv + clusters.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_derep_provirus_coords.tsv:md5,8c7a6441698648dfd73085524ce6a9cf"
                    ]
                ],
                "1": [
                    "versions.yml:md5,244bc446383cbd763a9b5a5d58208fb9"
                ],
                "derep_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_derep_provirus_coords.tsv:md5,8c7a6441698648dfd73085524ce6a9cf"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,244bc446383cbd763a9b5a5d58208fb9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T08:41:57.465382143"
    }
}